ansible_collections/arista/avd/roles/eos_designs/python_modules/sflow/avdstructuredconfig.py

```diff
@@ -0,0 +1,132 @@
+            return None
+
+        destinations = get(self._hostvars, "sflow_settings.destinations")
+        if destinations is None:
+            # TODO:
+            # AVD5.0.0 raise an error if sflow is enabled on an interface but there are no destinations configured.
+            # This cannot be implemented today since it would be breaking for already released support for sflow on interfaces.
+            return None
+
+        # At this point we have at least one interface with sFlow enabled
+        # and at least one destination.
+        sflow = {"run": True}
+
+        # Using a temporary dict for VRFs
+        sflow_vrfs = {}
+
+        for destination in natural_sort(destinations, "destination"):
+            vrf = get(destination, "vrf")
+            if vrf is None:
+                vrf = self.shared_utils.default_mgmt_protocol_vrf
+                source_interface = self.shared_utils.default_mgmt_protocol_interface
+
+            elif vrf == "use_mgmt_interface_vrf":
+                if (self.shared_utils.mgmt_ip is None) and (self.shared_utils.ipv6_mgmt_ip is None):
+                    raise AristaAvdMissingVariableError(
+                        "Unable to configure sFlow source-interface with 'use_mgmt_interface_vrf' since 'mgmt_ip' or 'ipv6_mgmt_ip' are not set."
+                    )
+
+                vrf = self.shared_utils.mgmt_interface_vrf
+                source_interface = self.shared_utils.mgmt_interface
+
+            elif vrf == "use_inband_mgmt_vrf":
+                # Check for missing interface
+                if self.shared_utils.inband_mgmt_interface is None:
+                    raise AristaAvdMissingVariableError(
+                        "Unable to configure sFlow source-interface with 'use_inband_mgmt_vrf' since 'inband_mgmt_interface' is not set."
+                    )
+
+                # self.shared_utils.inband_mgmt_vrf returns None for the default VRF, but here we need "default" to avoid duplicates.
+                vrf = self.shared_utils.inband_mgmt_vrf or "default"
+                source_interface = self.shared_utils.inband_mgmt_interface
+
+            if vrf in [None, "default"]:
+                # Add destination without VRF field
+                sflow.setdefault("destinations", []).append(
+                    {
+                        "destination": destination.get("destination"),
+                        "port": destination.get("port"),
+                    }
+                )
+                sflow["source_interface"] = source_interface
+
+            else:
+                # Add destination with VRF field.
+                sflow_vrfs.setdefault(vrf, {}).setdefault("destinations", []).append(
+                    {
+                        "destination": destination.get("destination"),
+                        "port": destination.get("port"),
+                    }
+                )
+                sflow_vrfs[vrf]["source_interface"] = source_interface
```

Comment:
Really not sure how I feel about relying on using `structured_config` to make the user set the `source_interface` for arbitrary VRFs.  For example I tried the following: 
```yaml
sflow_settings:
  destinations:
    - destination: 10.94.1.42
      vrf: MGMT
    - destination: 127.0.0.1
      vrf: default
  structured_config:
    source_interface: Loopback0
```
and got this error:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: UnboundLocalError: cannot access local variable 'source_interface' where it is not associated with a value
fatal: [dc1-leaf1b -> localhost]: FAILED! => {"changed": false, "msg": "cannot access local variable 'source_interface' where it is not associated with a value"}
```
It feels to me like it would be a nicer user experience to allow this to be overridden directly as part of this schema  and to raise a more informative error if this is mistakenly omitted.  The error above is not at all clear to the average user what they've done wrong.

Reply:
Re TerminAttr, I plan a follow up PR with a single knob to enable streaming to terminattr. It will auto-create a dummy interface in the default VRF, if you don't have one (like an l2leaf).

Reply:
@jonxstill I don't think source interface is required for sflow is it? If so we should indeed add a model to define arbitrary source interfaces and detect missing. If not I think we just need to fix this bug by ignoring.

Reply:
Finally fixed this. I decided to add a new `sflow_settings.vrfs` list to set/override source interface per VRF.